test/e2e/testdata/ext-proc-service.yaml

```diff
@@ -52,6 +52,21 @@ spec:
       labels:
         app: grpc-ext-proc
     spec:
+      affinity:
```

Comment:
why we need this?

Reply:
Because the UDS socket is only available on the local host. In order to test zone aware routing I've configured NUM_WORKERS=2 for e2e tests so the UDS pods need to run where the envoy proxy pods are.

That's what I was referring to here - https://github.com/envoyproxy/gateway/pull/5352#issuecomment-2832470367

Reply:
It is a bit confusing that this test case has been affected。

Reply:
Previously there was only a single node in the kind cluster so all pods were already guaranteed to run on the same node. When using a unix domain socket Backend that must be mounted to the envoy proxy pods and thus scheduled to the same node. Now that the number of nodes in the kind cluster has been increased we need to explicitly configure that.

Reply:
can you address these in the yaml?

Reply:
Isn't that already the requirement described in the docs? This should already be the default configuration but was only working because of the single node kind cluster.

Ref: https://gateway.envoyproxy.io/docs/tasks/traffic/backend/#restrictions
> Envoy Gateway does not manage the lifecycle of unix domain sockets referenced by the Backend resource. Envoy Gateway admins are responsible for creating and mounting the sockets into the envoy proxy pod. The latter can be achieved by patching the envoy deployment using the [EnvoyProxy](https://gateway.envoyproxy.io/docs/api/extension_types#envoyproxy) resource.

Reply:
can you address it in this test yaml?